contracts/programs/bubblegum/src/lib.rs

```diff
@@ -748,7 +755,19 @@ pub mod bubblegum {
                 metadata.symbol.clone(),
                 metadata.uri.clone(),
                 if metadata.creators.len() > 0 {
-                    Some(metadata.creators.iter().map(|c| c.adapt()).collect())
+                    let mut ammended_metadata_creators = metadata.creators;
+                    ammended_metadata_creators.push(Creator {
+                        address: ctx.accounts.mint_authority.key(),
+                        verified: true,
+                        share: 0,
+                    });
+                    assert!(ammended_metadata_creators.len() <= mpl_token_metadata::state::MAX_CREATOR_LIMIT, "Supplied metadata can have at most {} creators to leave space for mint_authority PDA creator", mpl_token_metadata::state::MAX_CREATOR_LIMIT-1);
```

Comment:
Also throw this error on `mint_v1`.

Reply:
good catch, I was checking in reverse lol. We don't even need this assertion at all, since we know that MetadataArgs is compliant with whatever we allowed in `mint_v1` via the hash assertion at the top of this instruction. I'll build an `assert_data_valid` for `mint_v1` that ensures everything is MPL decompress compliant upfront, and then all we need to do is append the PDA in `decompress_v1`